### PR TITLE
feat: Add JSDoc @component annotation support

### PR DESCRIPTION
**Pull request by okonet**
- [x] Enables documenting styled-components without additional wrappers
- [x] Update documentation
- [x] Add tests

### CODE REVIEW

**Comment by okonet** [comment]
Related to https://github.com/styled-components/styled-components/issues/945

File: docs/Thirdparties.md [hunk 1]
```diff
@@ -173,9 +173,11 @@ const SalmonButton = styled.button`
   color: snow;
 `;
 
-export default function Button({ children, ...props }) {
-  return <SalmonButton {...props}>{children}</SalmonButton>;
-}
+/**
+ * @component
+ */
+export default SalmonButton;
```
Comment (sapegin) [inline docs/Thirdparties.md]: Will `/** @component */` work too?

File: scripts/schemas/config.js [hunk 2]
```diff
@@ -107,7 +108,12 @@ module.exports = {
 	},
 	resolver: {
 		type: 'function',
-		default: reactDocgen.resolver.findAllExportedComponentDefinitions,
+		default: (ast, recast) => {
+			const defaultResolver = reactDocgen.resolver.findAllExportedComponentDefinitions;
```
Comment (sapegin) [inline scripts/schemas/config.js]: I’d not call it a default resolver, it’s not default for react-docgen and not default for Styleguidist anymore ;-)

**Comment by sapegin** [comment]
Would be nice to have some tests too ;-)

And thanks for the pull request!

File: scripts/schemas/config.js (hunk 2 above)
Comment (okonet) [reply scripts/schemas/config.js]: So what would you use instead?

File: docs/Thirdparties.md (hunk 1 above)
Comment (okonet) [reply docs/Thirdparties.md]: Yes, as long it's a valid JSDoc it will work. Tests are following.

File: scripts/schemas/config.js (hunk 2 above)
Comment (sapegin) [reply scripts/schemas/config.js]: I’d use `findAllExportedComponentDefinitions` probably, it’s clear at least ;-)

File: docs/Thirdparties.md (hunk 1 above)
Comment (sapegin) [reply docs/Thirdparties.md]: I’d use shorter form in the docs then ;-)

**Comment by MoeSattler** [comment]
@sapegin @okonet this is really important. Can I help to make this happen?

**Comment by sapegin** [comment]
@MoeSattler yes you can and it will be amazing! We need some basic tests to make sure we won’t break this.

**Comment by sapegin** [comment]
Probably something like this would work:

https://github.com/styleguidist/react-styleguidist/blob/12b8db1f5267fb147a9216b5e8cd8dc41a74447c/loaders/__tests__/props-loader.spec.js#L27-L44

**Comment by okonet** [comment]
@sapegin added tests and rebased (and renamed) commits on top of latest `master`

**Comment by sapegin** [comment]
Thanks, merged!

**Comment by MoeSattler** [comment]
Thanks a lot @okonet !
@sapegin any idea already when you will release?

**Comment by sapegin** [comment]
@MoeSattler Soonish, next week probably ;-)

**Pull request merged**